control-plane/cmd/post-install/kafka_channel_migrator.go

```diff
@@ -243,15 +250,90 @@ func (m *kafkaChannelMigrator) migrateConfigmap(ctx context.Context, logger *zap
 	return nil
 }
 
-func getEventingKafkaConfig(configMap map[string]string) (*config.EventingKafkaConfig, error) {
-	// Unmarshal The Eventing-Kafka ConfigMap YAML Into A EventingKafkaSettings Struct
-	eventingKafkaConfig := &config.EventingKafkaConfig{}
-	err := yaml.Unmarshal([]byte(configMap[constants.EventingKafkaSettingsConfigKey]), &eventingKafkaConfig)
+func (m *kafkaChannelMigrator) migrateSecret(ctx context.Context, logger *zap.SugaredLogger) error {
+	// Old configmap doesn't require an explicit specification of the protocol.
+	// Thus, we build the old config, see if TLS and SASL are enabled and set the protocol on the secret accordingly.
+	// By this approach, we touch the secret but we don't need to do some on-the-fly config upgrade in the channel
+	// reconciliation code.
+
+	logger.Infof("Migrating auth secret.")
+
+	oldcm, err := m.k8s.CoreV1().
```

Comment:
this logic seems redundant 


can we run instead just run the "patching" as a dedicatd function from where we already deal with the old config?